### PR TITLE
Spear Changes and Throwing Knife Speed Buff

### PR DESCRIPTION
**Pull request by assortedbeads**
## About The Pull Request

This PR expands the spear selection and brings them inline with other melee options.

New types of spears are; Glass, Steel, Plasteel and Uranium
All above types can be crafted, the first two types can be found in maintenance with varying rarity.

This PR also edits the throw speed of the newly added spears and the already available throwing knives, this new throw speed matches the throw speed of the NT Pilum.

In-hand, wielded, suit storage and back sprites were generously done by Rosalimo.
I had a great deal of assistance from Kirov with making sure the radiation effect on the Uranium spear worked, as I was having trouble with it.

## Why It's Good For The Game

I believe it's good for the game because it adds another melee option for people who enjoy melee combat, I am one of said people.

It additionally gives you  an active choice in what weapon you choose. 
- Bladed sword-esque weapons currently have the benefit of decapping/delimbing with decent penetration.
- Hammer type weapons have very high penetration.
- Spears have the benefit of a ranged attack to finish off a fleeing opponent,  or you can simply use them as you would a sword.

[Document showing changes](https://docs.google.com/spreadsheets/d/1YE5jRkzpAWpIUOUxmQ_8p7RxcJvaszR7gK6RsMOsQPM/edit?usp=sharing)

## Changelog
:cl:
balance: 
Rebalanced the damage of the old spear so it properly references our damage define.
Arranged the new spears in an increasing tier list, making sure none are as strong as the current NT spears.

imageadd: 
Added icons for the 4 types of spears, which show up inhand, wielded and on back and suit storage

code: 
Added recipes for crafting all the spears

code: 
Put the throw speed of the new spears and the already added throwing knife in line with current ranged options.
Gave the various spears style damage when thrown according to their tier, change in line with Pilum.
/:cl:

spear inhand 1
spear inhand
Spear Ground

### CODE REVIEW

File: code/datums/craft/recipes/weapon.dm [hunk 1]
```diff
@@ -210,6 +210,44 @@
...
+	name = "steel spear"
+	result = /obj/item/tool/spear/steel
+	steps = list(
+		list(/obj/item/stack/rods, 2, "time" = 30),
+		list(QUALITY_WELDING, 10, "time" = 30),
+		list(/obj/item/stack/cable_coil, 2, "time" = 10),
+		list(CRAFT_MATERIAL, 2, MATERIAL_STEEL, "time" = 10),
+		list(QUALITY_WELDING, 10, "time" = 30),
+		list(QUALITY_HAMMERING, 5, 10),
+	)
```
Comment (SirRichardFrancis) [inline code/datums/craft/recipes/weapon.dm:223]:
```suggestion
		list(QUALITY_HAMMERING, 5, 10))
```

File: code/datums/craft/recipes/weapon.dm [hunk 2]
```diff
@@ -210,6 +210,44 @@
...
+	name = "plasteel spear"
+	result = /obj/item/tool/spear/plasteel
+	steps = list(
+		list(/obj/item/stack/rods, 2, "time" = 30),
+		list(QUALITY_WELDING, 10, "time" = 30),
+		list(/obj/item/stack/cable_coil, 2, "time" = 10),
+		list(CRAFT_MATERIAL, 2, MATERIAL_PLASTEEL, "time" = 10),
+		list(QUALITY_WELDING, 10, "time" = 30),
+		list(QUALITY_HAMMERING, 5, 10),
+	)
```
Comment (SirRichardFrancis) [inline code/datums/craft/recipes/weapon.dm:235]:
```suggestion
		list(QUALITY_HAMMERING, 5, 10))
```

File: code/datums/craft/recipes/weapon.dm [hunk 3]
```diff
@@ -210,6 +210,44 @@
...
+	steps = list(
+		list(/obj/item/stack/rods, 2, "time" = 30),
+		list(QUALITY_WELDING, 10, "time" = 30),
+		list(/obj/item/stack/cable_coil, 2, "time" = 10),
+		list(CRAFT_MATERIAL, 2, MATERIAL_STEEL, "time" = 10),
+		list(QUALITY_WELDING, 10, "time" = 30),
+		list(CRAFT_MATERIAL, 1, MATERIAL_URANIUM, "time" = 10),
+		list(QUALITY_WELDING, 10, "time" = 30),
+		list(QUALITY_HAMMERING, 5, 10),
+	)
```
Comment (SirRichardFrancis) [inline code/datums/craft/recipes/weapon.dm:249]:
```suggestion
		list(QUALITY_HAMMERING, 5, 10))
```

File: code/game/objects/items/stacks/knife.dm [hunk 4]
```diff
@@ -78,6 +79,7 @@
 	force = WEAPON_FORCE_NORMAL
 	throwforce = WEAPON_FORCE_NORMAL
 	armor_penetration = ARMOR_PEN_SHALLOW
+	throw_speed = 3
```
Comment (SirRichardFrancis) [inline code/game/objects/items/stacks/knife.dm:82]:
```suggestion
```
Parent item have the same value, overriding is pointless here.

File: code/game/objects/items/weapons/tools/knives.dm [hunk 5]
```diff
@@ -325,29 +325,117 @@
...
 	allow_spin = FALSE
+	style_damage = 20
 
 	rarity_value = 20
 	spawn_tags = SPAWN_TAG_KNIFE
 
+/obj/item/tool/spear/steel
+	name = "steel spear"
+	desc = "A steel spearhead welded to a crude metal shaft, made from two welded rods. It'll serve well enough."
+	icon = 'icons/obj/weapons.dmi'
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:359]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 6]
```diff
@@ -325,29 +325,117 @@
...
+	icon_state = "spear_steel"
+	item_state = "spear_steel"
+	wielded_icon = "spear_steel_wielded"
+	flags = CONDUCT
+	sharp = TRUE
+	edge = TRUE
+	worksound = WORKSOUND_HARD_SLASH
+	w_class = ITEM_SIZE_BULKY //4 , it's a spear mate
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:367]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 7]
```diff
@@ -325,29 +325,117 @@
...
+	force = WEAPON_FORCE_DANGEROUS 
+	throwforce = WEAPON_FORCE_ROBUST 
+	armor_penetration = ARMOR_PEN_DEEP 
+	throw_speed = 3
+	max_upgrades = 5
+	tool_qualities = list(QUALITY_CUTTING = 10,  QUALITY_WIRE_CUTTING = 5, QUALITY_SCREW_DRIVING = 5)
+	matter = list(MATERIAL_STEEL = 3)
+	attack_verb = list("slashed", "stabbed") 
+	hitsound = 'sound/weapons/melee/heavystab.ogg'
+	slot_flags = SLOT_BACK
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:377]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 8]
```diff
@@ -325,29 +325,117 @@
...
+	structure_damage_factor = STRUCTURE_DAMAGE_WEAK
+	allow_spin = FALSE
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:379]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 9]
```diff
@@ -325,29 +325,117 @@
...
+	style_damage = 30
+
+	rarity_value = 60
+	spawn_tags = SPAWN_TAG_KNIFE
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:383]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 10]
```diff
@@ -325,29 +325,117 @@
...
+
+/obj/item/tool/spear/plasteel
+	name = "plasteel spear"
+	desc = "A carefully crafted plasteel spearhead affixed to a metal shaft, it is welded securely on and feels balanced. Show them the past still lives."
+	icon = 'icons/obj/weapons.dmi'
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:388]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 11]
```diff
@@ -325,29 +325,117 @@
...
+	icon_state = "spear_plasteel"
+	item_state = "spear_plasteel"
+	wielded_icon = "spear_plasteel_wielded"
+	flags = CONDUCT
+	sharp = TRUE
+	edge = TRUE
+	worksound = WORKSOUND_HARD_SLASH
+	w_class = ITEM_SIZE_BULKY
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:396]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 12]
```diff
@@ -325,29 +325,117 @@
...
+	flags = CONDUCT
+	sharp = TRUE
+	edge = TRUE
+	worksound = WORKSOUND_HARD_SLASH
+	w_class = ITEM_SIZE_BULKY 
+	force = WEAPON_FORCE_ROBUST 
+	throwforce = WEAPON_FORCE_BRUTAL 
+	armor_penetration = ARMOR_PEN_DEEP 
+	throw_speed = 3
+	max_upgrades = 5
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:401]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 13]
```diff
@@ -325,29 +325,117 @@
...
+	tool_qualities = list(QUALITY_CUTTING = 15,  QUALITY_WIRE_CUTTING = 10, QUALITY_SCREW_DRIVING = 10)
+	matter = list(MATERIAL_STEEL = 1, MATERIAL_PLASTEEL = 2)
+	attack_verb = list("slashed", "stabbed") 
+	hitsound = 'sound/weapons/melee/heavystab.ogg'
+	slot_flags = SLOT_BACK
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:406]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 14]
```diff
@@ -325,29 +325,117 @@
...
+	structure_damage_factor = STRUCTURE_DAMAGE_NORMAL
+	allow_spin = FALSE
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:408]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 15]
```diff
@@ -325,29 +325,117 @@
...
+	desc = "A steel spear with a uranium lined spearhead. Your foes may survive the stab, but the toxin will linger."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "spear_uranium"
+	item_state = "spear_uranium"
+	wielded_icon = "spear_uranium_wielded"
+	flags = CONDUCT
+	sharp = TRUE
+	edge = TRUE
+	worksound = WORKSOUND_HARD_SLASH
+	w_class = ITEM_SIZE_BULKY
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:422]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 16]
```diff
@@ -325,29 +325,117 @@
...
+	hitsound = 'sound/weapons/melee/heavystab.ogg'
+	slot_flags = SLOT_BACK
+	structure_damage_factor = STRUCTURE_DAMAGE_NORMAL
+	allow_spin = FALSE
+	style_damage = 50
+
+/obj/item/tool/spear/uranium
+	name = "uranium spear"
+	desc = "A steel spear with a uranium lined spearhead. Your foes may survive the stab, but the toxin will linger."
+	icon = 'icons/obj/weapons.dmi'
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:414]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 17]
```diff
@@ -325,29 +325,117 @@
...
+	flags = CONDUCT
+	sharp = TRUE
+	edge = TRUE
+	worksound = WORKSOUND_HARD_SLASH
+	w_class = ITEM_SIZE_BULKY //4 , it's a spear mate
+	force = WEAPON_FORCE_DANGEROUS 
+	throwforce = WEAPON_FORCE_ROBUST 
+	armor_penetration = ARMOR_PEN_DEEP 
+	throw_speed = 3
+	max_upgrades = 5
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:372]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 18]
```diff
@@ -325,29 +325,117 @@
...
+	flags = CONDUCT
+	sharp = TRUE
+	edge = TRUE
+	worksound = WORKSOUND_HARD_SLASH
+	w_class = ITEM_SIZE_BULKY 
+	force = WEAPON_FORCE_DANGEROUS 
+	throwforce = WEAPON_FORCE_DANGEROUS 
+	armor_penetration = ARMOR_PEN_DEEP 
+	throw_speed = 3
+	max_upgrades = 5
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:427]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 19]
```diff
@@ -325,29 +325,117 @@
...
+	tool_qualities = list(QUALITY_CUTTING = 10,  QUALITY_WIRE_CUTTING = 5, QUALITY_SCREW_DRIVING = 5)
+	matter = list(MATERIAL_STEEL = 3, MATERIAL_URANIUM = 1)
+	attack_verb = list("slashed", "stabbed") 
+	hitsound = 'sound/weapons/melee/heavystab.ogg'
+	slot_flags = SLOT_BACK
+	structure_damage_factor = STRUCTURE_DAMAGE_BLADE
+	allow_spin = FALSE
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:434]:
```suggestion
```

File: code/game/objects/items/weapons/tools/knives.dm [hunk 20]
```diff
@@ -325,29 +325,117 @@
...
+	style_damage = 50
+
+/obj/item/tool/spear/uranium/apply_hit_effect(mob/living/target, mob/living/user, hit_zone)
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/tools/knives.dm:437]:
```suggestion
/obj/item/tool/spear/uranium/apply_hit_effect(mob/living/carbon/human/target, mob/living/user, hit_zone)
```

**Review by SirRichardFrancis — CHANGES_REQUESTED** [review]

**Review by hyperioo — APPROVED** [review]

**Pull request merged**